### PR TITLE
Added ability to load Swagger JSON file into objects

### PR DESCRIPTION
**Pull request by geoffsmith82**
Hi Marcelo,

I have added the ability to load a Swagger JSON file into the objects as well.  I have also added in a few extra properties on several of your objects.  I have also included a very simple demo of loading a swagger file.

### CODE REVIEW

**Review by marcelojaloto — COMMENTED** [review]
@geoffsmith82,
Thank you for your collaboration.

**Comment by marcelojaloto** [comment]
@cunhaw 
Could you please also check?

File: Source/Swag.Doc.Info.pas [hunk 1]
```diff
@@ -35,14 +35,17 @@   TSwagInfo = class(TObject)
     fTitle: string;
     fDescription: string;
     fContact: TSwagInfoContact;
+    FTermsOfService: string;
```
Comment (cunhaw) [inline Source/Swag.Doc.Info.pas]: fTermsOfService

File: Source/Swag.Doc.Info.pas [hunk 2]
```diff
@@ -78,7 +100,11 @@ function TSwagInfo.GenerateJsonObject: TJSONObject;
   Result.AddPair(c_SwagInfoVersion, fVersion);
   Result.AddPair(c_SwagInfoTitle, fTitle);
   Result.AddPair(c_SwagInfoDescription, fDescription);
-  Result.AddPair(c_SwagInfoContact, fContact.GenerateJsonObject);
+  if not FTermsOfService.IsEmpty then
```
Comment (cunhaw) [inline Source/Swag.Doc.Info.pas]: fTermsOfService

File: Source/Swag.Doc.Info.pas [hunk 3]
```diff
@@ -35,14 +35,17 @@   TSwagInfo = class(TObject)
...
   public
     constructor Create; reintroduce;
     destructor Destroy; override;
 
     function GenerateJsonObject: TJSONObject;
+    procedure Load(pJson: TJSONObject);
 
     property Version: string read fVersion write fVersion;
     property Title: string read fTitle write fTitle;
+    property TermsOfService: string read FTermsOfService write FTermsOfService;
```
Comment (cunhaw) [inline Source/Swag.Doc.Info.pas]: fTermsOfService

**Review by cunhaw — CHANGES_REQUESTED** [review]

File: Source/Swag.Doc.pas [hunk 4]
```diff
@@ -246,6 +259,56 @@ function TSwagDoc.GetSwaggerVersion: string;
...
+
+  fLicense.Load((fSwaggerJson as TJSONObject).Values[c_SwagLicense] as TJSONObject);
+
+  vJsonObj := (fSwaggerJson as TJSONObject).Values[c_SwagPaths] as TJSONObject;
+  vJsonSchemesArray := (fSwaggerJson as TJSONObject).Values[c_SwagSchemes] as TJSONArray;
+
+  for j := 0 to vJsonSchemesArray.Count - 1 do
+  begin
+    if (vJsonSchemesArray.Items[j].Value = 'http') then
+      Schemes := self.Schemes + [tpsHttp]
```
Comment (cunhaw) [inline Source/Swag.Doc.pas]: Self

File: Source/Swag.Doc.pas [hunk 5]
```diff
@@ -246,6 +259,56 @@ function TSwagDoc.GetSwaggerVersion: string;
...
+    else if (vJsonSchemesArray.Items[j].Value = 'https') then
+      Schemes := self.Schemes + [tpsHttps];
```
Comment (cunhaw) [inline Source/Swag.Doc.pas]: Self

File: Source/Swag.Doc.pas [hunk 6]
```diff
@@ -67,11 +69,14 @@   TSwagDoc = class(TObject)
...
     procedure SaveSwaggerJsonToFile;
 
+    procedure LoadFromFile(const pFilename: string);
+
     property SwaggerFilesFolder: string read fSwaggerFilesFolder write SetSwaggerFilesFolder;
     property SwaggerJson: TJSONValue read fSwaggerJson;
 
     property SwaggerVersion: string read GetSwaggerVersion;
     property Info: TSwagInfo read fInfo;
+    property License : TSwagInfoLicense read fLicense;
```
Comment (cunhaw) [inline Source/Swag.Doc.pas]: License: TSwagInfoLicense (remove blank space)

File: Source/Swag.Doc.pas [hunk 7]
```diff
@@ -38,6 +39,7 @@ interface
   TSwagDoc = class(TObject)
   private
     fInfo: TSwagInfo;
+    fLicense : TSwagInfoLicense;
```
Comment (cunhaw) [inline Source/Swag.Doc.pas]: fLicense: TSwagInfoLicense; (remove blank space)

**Review by cunhaw — CHANGES_REQUESTED** [review]
Just small formatting issues

**Comment by geoffsmith82** [comment]
After you have merged this PR in, I have a demo of generating a swagger file by code and some more of the swagger spec implemented.

File: Source/Swag.Doc.pas [hunk 8]
```diff
@@ -279,10 +279,14 @@ procedure TSwagDoc.LoadFromFile(const pFilename: string);
 
   for j := 0 to vJsonSchemesArray.Count - 1 do
   begin
-    if (vJsonSchemesArray.Items[j].Value = 'http') then
```
Comment (marcelojaloto) [inline Source/Swag.Doc.pas]: See the improvements in unit Swag.Common.Types.Helpers.

**Review by marcelojaloto — APPROVED** [review]
See improvements and code pattern changes to follow in your next changes.

**Comment by marcelojaloto** [comment]
@geoffsmith82, 
Please make another PR for your new collaborations. Thanks!

**Pull request merged**